src/main/java/nextstep/app/application/OAuth2TokenStrategyRequester.java

```diff
@@ -0,0 +1,62 @@
+package nextstep.app.application;
+
+import nextstep.security.authentication.AuthenticationException;
+import nextstep.security.authentication.OAuth2TokenRequestStrategy;
+import nextstep.security.authentication.TokenRequest;
+import nextstep.security.authentication.TokenResponse;
+import nextstep.security.authentication.OAuth2TokenRequester;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Service
+public class OAuth2TokenStrategyRequester implements OAuth2TokenRequester {
+
+    private static final Logger log = LoggerFactory.getLogger(OAuth2TokenStrategyRequester.class);
+
+    private final RestTemplate restTemplate;
+    private final Map<String, OAuth2TokenRequestStrategy> strategies;
+
+    public OAuth2TokenStrategyRequester(RestTemplate restTemplate, List<OAuth2TokenRequestStrategy> strategies) {
+        this.restTemplate = restTemplate;
+        this.strategies = strategies.stream()
+                .collect(Collectors.toMap(
+                        OAuth2TokenRequestStrategy::getOAuth2Type,
+                        strategy -> strategy
+                ));
```

Comment:
생성 이후 변경될 필요가 없으므로, 불변(immutable)으로 관리하는 것도 좋겠네요 😄 
```suggestion
                this.strategies = strategies.stream()
                .collect(Collectors.toUnmodifiableMap(
                        OAuth2TokenRequestStrategy::getOAuth2Type,
                        strategy -> strategy
                ));
```